File: BE-Ophelia/BE-Ophelia/Models/Compra.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BE_Ophelia.Models
{
    public class Compra
    {
        [Key]
        public int compraId { get; set; }
        [Required]
        public int cantidad { get; set; }
        [Required]
        public int precioPagado { get; set; }
        public int productoId { get; set; }
        [ForeignKey("productoId")]
        public Producto cliente { get; set; }

        public int facturaId { get; set; }
        [ForeignKey("facturaId")]
        public Factura factura { get; set; }
    }
}

File: Angular/FE-Ophelia/src/app/interface/compra.ts
import { Factura } from "./factura";
import { Producto } from "./producto";

export interface Compra {
    compraId?: number,
    precioPagado: number,
    cantidad: number,
    facturaId: number,
    productoId: number,
    factura: Factura,
    producto: Producto
}
